### ISSUE TITLE
Public C interface -- non-int32 data types

### ISSUE THREAD

**Issue by yvdriess** [gunrock/gunrock#291]
Some graph benchmarks require larger datatypes to be used for labels and indices. For example, graph500 specifies vertex labels to allocate at least 48 bits.

The public C interface appears to only have implemented a single VTXID_TYPE: VTXID_INT. (This specializes in a large switch statement eventually to an int32, but the bfs() interface is hardcoded for int32 labels) The CUDA-portion of Gunrock does appears to have painstakenly templated datatypes to support differentvertex/edge/weight/size datatypes.

Are there any plans for adding larger vertex label datatypes to the public interface?

**Comment by jowens** [comment]
In the short term, would just using the lower-level-not-public-C-interface Gunrock work for you?

**Comment by yvdriess** [comment]
Definitely, although I am not sure what you are refering to. I couldn't find a lower-level C interface.
bfs() and gunrock_bfs() directly call into a CUDA dispatch_bfs() function with a switch statement that only implements the int32 cases.

**Comment by jowens** [comment]
OK. Need one of the students to chime in then b/c I don't want to give you bad advice. :) @sgpyc ? (But he's at SC this week.) @neoblizz ?

**Comment by sgpyc** [comment]
We do have plan to support more data types in the C-public interfaces (or C++, if full template support is needed).
The Gunrock internals are fully templatlized, we have tested 32-bit and 64-bit integers for vertex ids and edge ids, 32-bit and 64-bit floating points for edge values etc. But for non-standard datatypes, such as 24-bit integers, we may need some more efforts to verify our code and to optimize some underlaying memory I/O routines.

If needed in short time, https://github.com/gunrock/gunrock/blob/master/gunrock/app/bfs/bfs_app.cu#L297 is the 32-bit public C API warper, and another switch statement for 64-bit can be added. It should work, just make sure data past in are correctly typed.

**Comment by yvdriess** [comment]
Thank you, I will make the changes and issue a pull request.

**Comment by neoblizz** [comment]
@yvdriess let us know if you tried anything, we'd love to add that in if possible. :)
